### PR TITLE
upgrade to ts 4.5.2

### PR DESCRIPTION
**Pull request by bluebill1049**
- update to 4.5.0-dev
- build pass
- revert b83fb02

### CODE REVIEW

**Comment by bluebill1049** [comment]
looks like 4.5 is no longer working with ✨ enable sharing form sections between forms in typescript (#6807)

File: src/__tests__/type.test.tsx [hunk 1]
```diff
@@ -27,21 +27,15 @@ test('should not throw type error with path name', () => {
...
     foo: string;
     bar: 'test1' | 'test2';
     test2: boolean;
-  } & FormFieldsSmall;
+  };
 
   function App() {
     const form = useForm<FormFieldsBig>();
-    const test: Omit<UseFormReturn<FormFieldsSmall>, 'watch'> = form;
+    const test: Omit<UseFormReturn<FormFieldsBig>, 'watch'> = form;
```
Comment (bluebill1049) [inline src/__tests__/type.test.tsx:38]: result in type error

**Comment by bluebill1049** [comment]
Made some update let me know your thoughts @barrymay @kotarella1110.

**Comment by barrymay** [comment]
Excellent work Bill well done!  It's great to see that simplify clarifying the passed down types fixed the build and the tests!

File: src/__tests__/type.test.tsx (hunk 1 above)
Comment (barrymay) [reply src/__tests__/type.test.tsx:38]: In this one I would really revert the code and (maybe) comment out the test.  The original code was really what @SimonSelg (understandably) wanted, whereas changing FormFieldsSmall to FormFieldsBig no longer lets the test agree with tits own description.

Comment (bluebill1049) [reply src/__tests__/type.test.tsx:38]: yea, we will have to revert it because its not doing what it desired anymore.

**Review by barrymay — DISMISSED** [review]

**Comment by barrymay** [comment]
Acknowledging our need to discuss the internal 'as' changes (which we'll do!) and type clarifications (and the need to investigate tsd to support testing the types), I agree this is good to go for now.   (to be clear, I wouldn't deploy this until we can run fine on 4.5 final, but it's good otherwise)

**Comment by bluebill1049** [comment]
yea @barrymay, I am keeping this feature branch running until 4.5 is released.

**Pull request merged**